src/plone/distribution/handler.py

```diff
@@ -35,13 +37,14 @@ def default_handler(
         # Process content import from json
         content_json_path = contents["json"]
         if content_json_path:
-            # If there is no savepoint most tests fail with a PosKeyError
-            transaction.savepoint(optimistic=True)
             # Invalidate the schema cache to make sure we get up to date behaviors.
             # Normally this happens on commit, but we didn't commit yet.
             SCHEMA_CACHE.clear()
             importer = get_importer(site)
             importer.import_site(content_json_path)
+            # Create a savepoint to ensure the import is atomic
+            transaction.savepoint(optimistic=True)
+            # Commit the transaction to finalize the import
```

Comment:
It seems strange to me to create a savepoint immediately before the commit. Does that actually do anything different from simply committing?